src/picongpu/include/fields/background/templates/TWTS/BField.tpp

```diff
@@ -163,6 +196,56 @@ namespace twts
                          float_X(0.0) );
     }
 
+    template<>
+    HDINLINE float3_X
+    BField::getTWTSBfield_Normalized_Ey<DIM2>(
+            const PMacc::math::Vector<floatD_64,detail::numComponents>& bFieldPositions_SI,
+            const float_64 time) const
+    {
+        typedef PMacc::math::Vector<float3_64,detail::numComponents> PosVecVec;
+        PosVecVec pos(PosVecVec::create(
+                                           float3_64::create(0.0)
+                                       ));
+
+        for (uint32_t k = 0; k<detail::numComponents;++k) {
+            /* The 2D output of getFieldPositions_SI only returns
+             * the y- and z-component of a 3D vector. */
+            for (uint32_t i = 0; i<simDim;++i) pos[k][i+1] = bFieldPositions_SI[k][i];
+        }
+        /*  Corresponding position vector for the Field-components in 2D simulations.
```

Comment:
Might be that this comment block should be moved to the method/function description part over the method.